### PR TITLE
Fix GDExtension classes inheriting from Resource not using icons in FileSystem

### LINKED ISSUE #97387: GDExtension classes inheriting from Resource are not using icons in FileSystem
### Tested versions

4.3 dev 1

### System information

Windows 10 Godot 4.3 dev 1 Vulkan

### Issue description

I am aware there is https://github.com/godotengine/godot/issues/32706, but it does not seem to cover GDExtensions.

Currently, GDExtension classes inheriting from Resource are not using icons in FileSystem. It seems to work in every other place, but not in FileSystem for some reason.

image
image
image

### Steps to reproduce

1. Create a class in GDExtension that inherits Resource
2. Assign an icon in .gdextension file to said class
3. Observe how FileSystem is using default icon for this class, while the property viewer is displaying everything as expected

### Minimal reproduction project (MRP)

You can see an example in the root of the project here
image
[IconTest.zip](https://github.com/user-attachments/files/17106236/IconTest.zip)

### PR DESCRIPTION
**Pull request by SaNeOr**
fix #97387

### CODE REVIEW

**Review by Hilderin — APPROVED** [review]
Sounds good to me! Tested with the MRP on master and with this PR and it fixes the issue as expected.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
This seems to cause a crash in one of my test projects:

```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.4.dev.custom_build (ac1ef5c9f64f1c21f9addfc98be460e08dff8fb7)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib64/libc.so.6(+0x40d00) [0x7efce506ad00] (??:0)
[2] CowData<char32_t>::_ref(CowData<char32_t> const&) (/home/akien/Godot/godot/./core/templates/cowdata.h:461)
[3] String::String(String const&) (/home/akien/Godot/godot/./core/string/ustring.h:483)
[4] Resource::get_path() const (/home/akien/Godot/godot/./core/io/resource.cpp:95)
[5] EditorFileSystem::_update_file_icon_path(EditorFileSystemDirectory::FileInfo*) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1981 (discriminator 6))
[6] EditorFileSystem::_update_files_icon_path(EditorFileSystemDirectory*) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1995 (discriminator 3))
[7] EditorFileSystem::_update_files_icon_path(EditorFileSystemDirectory*) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1992 (discriminator 3))
[8] EditorFileSystem::_update_files_icon_path(EditorFileSystemDirectory*) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1992 (discriminator 3))
[9] EditorFileSystem::_update_files_icon_path(EditorFileSystemDirectory*) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1992 (discriminator 3))
[10] EditorFileSystem::_notification(int) (/home/akien/Godot/godot/./editor/editor_file_system.cpp:1672)
[11] EditorFileSystem::_notificationv(int, bool) (/home/akien/Godot/godot/./editor/editor_file_system.h:145 (discriminator 14))
[12] Object::notification(int, bool) (/home/akien/Godot/godot/./core/object/object.cpp:878)
[13] SceneTree::_process_group(SceneTree::ProcessGroup*, bool) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:1020)
[14] SceneTree::_process(bool) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:1092 (discriminator 2))
[15] SceneTree::process(double) (/home/akien/Godot/godot/./scene/main/scene_tree.cpp:582)
[16] Main::iteration() (/home/akien/Godot/godot/main/main.cpp:4383 (discriminator 3))
[17] OS_LinuxBSD::run() (/home/akien/Godot/godot/platform/linuxbsd/os_linuxbsd.cpp:962 (discriminator 1))
[18] /home/akien/Godot/godot/bin/godot.linuxbsd.editor.dev.x86_64(main+0x14b) [0x6098af1] (/home/akien/Godot/godot/platform/linuxbsd/godot_linuxbsd.cpp:85)
[19] /lib64/libc.so.6(+0x2a088) [0x7efce5054088] (??:0)
[20] /lib64/libc.so.6(__libc_start_main+0x8b) [0x7efce505414b] (??:0)
[21] /home/akien/Godot/godot/bin/godot.linuxbsd.editor.dev.x86_64(_start+0x25) [0x60988e5] (??:?)
-- END OF BACKTRACE --
================================================================
```

Trying to figure out why and make a MRP.

**Comment by akien-mga** [comment]
It's crashing for me when there's an `.icns` icon in the project:
[new-test-project.zip](https://github.com/user-attachments/files/17148707/new-test-project.zip)

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -1872,6 +1872,10 @@ void EditorFileSystem::_update_file_icon_path(EditorFileSystemDirectory::FileInf
 		}
 	}
 
+	if (icon_path.is_empty() && !file_info->type.is_empty()) {
+		icon_path = EditorNode::get_singleton()->get_class_icon(file_info->type)->get_path();
```
Comment (akien-mga) [inline editor/editor_file_system.cpp:1876]: This is wrongly assuming that the result of `EditorNode::get_singleton()->get_class_icon(file_info->type)` is valid and can de-referenced.

Comment (akien-mga) [reply editor/editor_file_system.cpp:1876]: Fixed by #97491.

**Pull request merged**